nncf/tensorflow/api/compression.py

```diff
@@ -92,3 +96,50 @@ def apply_to(self, model: ModelType) -> ModelType:
             self.initialize(transformed_model)
 
         return transformed_model
+
+
+class TFCompressionState(CompressionState, tf.train.experimental.PythonState):
+    def __init__(self, compression_ctrl: TFCompressionAlgorithmController = None, builder_state: Dict = None):
+        self._compression_ctrl = compression_ctrl
+        self._builder_state = builder_state
+        self._ctrl_state = None
+
+    @property
+    def builder_state(self) -> Dict:
+        return self._builder_state
+
+    @property
+    def ctrl_state(self) -> Dict:
+        if self._compression_ctrl is not None:
+            return self._compression_ctrl.get_state()
+        return self._ctrl_state
+
+    def load_state(self, state: Dict):
+        """
+        Initializes object from the state.
+
+        :param state: Output of `get_state()` method.
+        """
+        self._builder_state = state[self.BUILDER_STATE]
+        self._ctrl_state = state[self.CONTROLLER_STATE]
+
+    def serialize(self) -> str:
+        """
+        Callback to serialize the object by tf.train.experimental.PythonState.
+
+        :return: Serialized compression state.
+        """
+        string_value = json.dumps(self.get_state())
+        return string_value
+
+    def deserialize(self, state: str) -> None:
+        """
+        Callback to deserialize the object by tf.train.experimental.PythonState.
+
+        :param state: Serialized compression state.
+        """
+        state = json.loads(state)
+        self.load_state(state)
+
+    def __bool__(self):
+        return self._ctrl_state is None and self._compression_ctrl is None
```

Comment:
```suggestion
        return self.ctrl_state is not None and self.builder_state is not None
```

Reply:
done, moved __bool__ to a common class and identically handled it `create_compressed_model` for TF & PT